### ISSUE TITLE
ObjectPropertyNaming defaults do not match Kotlin style guide

### ISSUE THREAD

**Issue by kevinherron** [arturbosch/detekt#1331]
## Expected Behavior
I expect code that follows the Kotlin style guide to not get flagged by detekt.

## Observed Behavior
Code like this:
```kotlin
    companion object {

        private val DEVICE_NAMESPACE_INDEX = ushort(1)

        ...
```

is being flagged by the ObjectPropertyNaming rule even though the style guide encourages this naming convention for immutable data: https://kotlinlang.org/docs/reference/coding-conventions.html#property-names.

## Your Environment
* Version of detekt used: 1.0.0-RC10

**Comment by kevinherron** [comment]
Hmm. detekt flags these because they are private. The style guide makes no explicit distinction either way, it just says "top-level or object val properties with no custom get function".

Not sure what the intention of the style guide is here. In Java it's not strange to make the constants uppercase whether it's `private static final` or `public static final`.

**Comment by vanniktech** [comment]
This is a regression. I belive. Didn't happen before RC10.

**Comment by schalkms** [comment]
Commit 17ea132 changed this behavior

**Comment by schalkms** [comment]
Seems to be a merge problem. I'll change it back and add a test
